### PR TITLE
Don't retry on some HTTP client errors

### PR DESCRIPTION
**Pull request by stevededalus**
- Don't retry on 402, 403, and 404 (Payment required, access forbidden, and not found) errors: they're unlikely to be transient.

- Drop retry count significantly: the purpose of the queue is to handle the majority of the retry logic

- Drop timeout based on context: if the lambda execution environment kills a process that's fine

- Reraise exception in tenacity decorators: right now they're raising a retry exception instead of the underlying exception.

### CODE REVIEW

File: lambdas/es/indexer/index.py [hunk 1]
```diff
@@ -487,13 +487,19 @@ def retry_s3(
     else:
         arguments['IfMatch'] = etag
 
-    time_remaining = get_time_remaining(context)
+    def not_known_exception(exception):
+        error_code = exception.response.get(['Error'], {}).get(['Code'], 218)
+        return error_code not in ["402", "403", "404"]
```
Comment (dimaryaz) [inline lambdas/es/indexer/index.py]: What error codes do we expect to see and retry? In general, no 4xx codes should be re-tried...

Comment (akarve) [reply lambdas/es/indexer/index.py]: That seems a bit extreme. e.g. we want to retry 429s (too many requests).

Comment (akarve) [reply lambdas/es/indexer/index.py]: Seems more correct to enumerate the 400s we do want to retry (it's true most aren't worth trying) there are only a few to retry; see Client Errors https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: lambdas/es/indexer/index.py [hunk 2]
```diff
@@ -487,13 +487,19 @@ def retry_s3(
     else:
         arguments['IfMatch'] = etag
 
-    time_remaining = get_time_remaining(context)
```
Comment (akarve) [inline lambdas/es/indexer/index.py]: should also remove `context` argument to this function

File: lambdas/es/indexer/index.py [hunk 3]
```diff
@@ -34,7 +34,7 @@
 CHUNK_LIMIT_BYTES = 20_000_000
 DOC_LIMIT_BYTES = 2_000
 ELASTIC_TIMEOUT = 30
-MAX_RETRY = 10 # prevent long-running lambdas due to malformed calls
+MAX_RETRY = 4 # prevent long-running lambdas due to malformed calls
```
Comment (akarve) [inline lambdas/es/indexer/index.py]: You can drop `RETRY_429` as well, for the same reasons. Even `RETRY_429 = MAX_RETRY`.

File: lambdas/es/indexer/index.py (hunk 1 above)
Comment (akarve) [reply lambdas/es/indexer/index.py]: I think 408, 409, and 429 are the only ones worth retrying.

File: lambdas/es/indexer/index.py (hunk 2 above)
Comment (stevededalus) [reply lambdas/es/indexer/index.py]: Done.

**Comment by stevededalus** [comment]
I'm going to keep this PR smaller in scope, can revisit playing the 429 retries and more restrictive error code lists in the future

**Pull request merged**